src/endpoint.rs

```diff
@@ -107,6 +107,18 @@ impl<B: UsbBus, D: EndpointDirection> Endpoint<'_, B, D> {
     }
 }
 
+impl<B: UsbBusEndpointStatus> Endpoint<'_, B, In> {
+    /// Returns true if the endpoint is ready to write another packet, otherwise returns false.
+    pub fn can_write(&self) -> bool {
+        self.bus().can_write(self.address).expect("unexpected error")
```

Comment:
I think it would be better to return errors and not panic. As you can see the built-in endpoint objects are kind of becoming dumb wrappers for `UsbBus` which is why I'm considering getting rid of them.